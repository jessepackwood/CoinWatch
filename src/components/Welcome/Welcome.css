@import url("https://fonts.googleapis.com/css?family=Abel");
.welcome {
  background: #244968;
  height: 100vh;
  width: 100vw;
  display: flex; }

.welcome-text {
  display: flex;
  color: #fff;
  position: absolute;
  padding: 120px;
  align-items: center; }

.welcome-title {
  font-family: "Abel", sans-serif;
  font-size: 42pt; }

.line-graph {
  padding-top: 380px;
  width: 100%;
  max-height: 100%; }

.start-button {
  background: none;
  color: #fff;
  width: 200px;
  height: 200px;
  margin-left: 200px;
  border: 12px solid white;
  border-radius: 100px;
  font-size: 23pt;
  outline: none; }
  .start-button:hover {
    border: 12px solid #9af765;
    cursor: pointer; }

.welcome-subtitle {
  color: #f2bb30; }
