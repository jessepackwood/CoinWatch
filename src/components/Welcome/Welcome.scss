@import '../../variable.scss';

.welcome {
	background: $blue;
	height: 100vh;
	width: 100vw;
	display: flex;
}

.welcome-text {
	display: flex;
	color: #fff;
	position: absolute;
	padding: 120px;
	align-items: center;
}
.welcome-title {
	font-family: $title-font;
	font-size: 42pt;
}

.line-graph {
	padding-top: 380px;
	width: 100%;
	max-height: 100%;
}

.start-button {
	background: none;
	color: #fff;
	width: 200px;
	height: 200px;
	margin-left: 200px;
	border: 12px solid white;
	border-radius: 100px;
	font-size: 23pt;
	outline: none;
	&:hover {
		border: 12px solid $green;
		cursor: pointer;
	}
}

.welcome-subtitle {
	color: $orange;
}