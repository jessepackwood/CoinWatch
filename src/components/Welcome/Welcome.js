import React from 'react'
import { Link } from 'react-router-dom'
import './Welcome.css'

const Welcome = () => {
	return (
		<div className='welcome'>
			<div className='welcome-text'>
				<div>
					<h1 className='welcome-title'>Coin Watch</h1>
					<p>Track your returns and only watch the coins you care about.</p>
				</div>
				<div>
					<Link to='/home'>
						<button className='start-button'>Get started</button>
					</Link>
				</div>
			</div>
			<img src={require('../../assets/thin-line.svg')} className='line-graph' />
		</div>
		)
}

export default Welcome 