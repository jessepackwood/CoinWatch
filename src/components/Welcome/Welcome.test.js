import React from 'react'
import { shallow } from 'enzyme'
import Welcome from './Welcome'

describe('Welcome tests', () => {
  const welcome = shallow(<Welcome />)

  it('should be defined', () => {
    expect(welcome).toBeDefined()
  })

  it('should match the snapshot', () => {
    expect(welcome).toMatchSnapshot()
  })

})