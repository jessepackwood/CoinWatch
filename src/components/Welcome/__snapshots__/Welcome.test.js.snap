// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="welcome-text"
>
          <div>
                    <h1
                              className="welcome-title"
                    >
                              Coin Watch
                    </h1>
                    <p>
                              Track your profits and only watch the coins you care about.
                    </p>
          </div>
          <div>
                    <Link
                              replace={false}
                              to="/login"
                    >
                              <button
                                        className="start-button"
                              >
                                        Get started
                              </button>
                    </Link>
          </div>
</div>,
        <img
          alt="line graph"
          className="line-graph"
          src="thin-line.svg"
/>,
      ],
      "className": "welcome",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h1
                            className="welcome-title"
              >
                            Coin Watch
              </h1>
              <p>
                            Track your profits and only watch the coins you care about.
              </p>
</div>,
            <div>
              <Link
                            replace={false}
                            to="/login"
              >
                            <button
                                          className="start-button"
                            >
                                          Get started
                            </button>
              </Link>
</div>,
          ],
          "className": "welcome-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="welcome-title"
>
                  Coin Watch
</h1>,
                <p>
                  Track your profits and only watch the coins you care about.
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Coin Watch",
                  "className": "welcome-title",
                },
                "ref": null,
                "rendered": "Coin Watch",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Track your profits and only watch the coins you care about.",
                },
                "ref": null,
                "rendered": "Track your profits and only watch the coins you care about.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/login"
>
                <button
                                className="start-button"
                >
                                Get started
                </button>
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="start-button"
>
                  Get started
</button>,
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get started",
                  "className": "start-button",
                },
                "ref": null,
                "rendered": "Get started",
                "type": "button",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "line graph",
          "className": "line-graph",
          "src": "thin-line.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="welcome-text"
>
            <div>
                        <h1
                                    className="welcome-title"
                        >
                                    Coin Watch
                        </h1>
                        <p>
                                    Track your profits and only watch the coins you care about.
                        </p>
            </div>
            <div>
                        <Link
                                    replace={false}
                                    to="/login"
                        >
                                    <button
                                                className="start-button"
                                    >
                                                Get started
                                    </button>
                        </Link>
            </div>
</div>,
          <img
            alt="line graph"
            className="line-graph"
            src="thin-line.svg"
/>,
        ],
        "className": "welcome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h1
                                className="welcome-title"
                >
                                Coin Watch
                </h1>
                <p>
                                Track your profits and only watch the coins you care about.
                </p>
</div>,
              <div>
                <Link
                                replace={false}
                                to="/login"
                >
                                <button
                                                className="start-button"
                                >
                                                Get started
                                </button>
                </Link>
</div>,
            ],
            "className": "welcome-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="welcome-title"
>
                    Coin Watch
</h1>,
                  <p>
                    Track your profits and only watch the coins you care about.
</p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Coin Watch",
                    "className": "welcome-title",
                  },
                  "ref": null,
                  "rendered": "Coin Watch",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Track your profits and only watch the coins you care about.",
                  },
                  "ref": null,
                  "rendered": "Track your profits and only watch the coins you care about.",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/login"
>
                  <button
                                    className="start-button"
                  >
                                    Get started
                  </button>
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="start-button"
>
                    Get started
</button>,
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Get started",
                    "className": "start-button",
                  },
                  "ref": null,
                  "rendered": "Get started",
                  "type": "button",
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "line graph",
            "className": "line-graph",
            "src": "thin-line.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
